Plans: Update storage limit for Business plan

jeremeylduvall opened the pull request:
Currently, the storage features at `plans/my-plan/site` are a bit confusing for Business customers. If you have the Business plan, you actually have unlimited storage, but the plan features page reads the Premium description ('The Premium plan also adds 10GB of file storage.'). This updates the features page to reflect the unlimited storage and modify the wording depending on the plan.

### To test
1. Load up this branch on an account that has at least a Business plan and Premium plan.
2. For each site, visit http://calypso.localhost:3000/plans/my-plan/. You should notice the correct storage limit (unlimited for Business, 10GB for Premium, nonexistent for Personal) under the "Video and audio posts" header.

### Screenshots

**Business**

<img width="482" alt="screen shot 2017-12-15 at 1 59 34 pm" src="https://user-images.githubusercontent.com/7240478/34060452-12ffc854-e1a1-11e7-965f-c7ebd33c522a.png">

**Premium**

<img width="474" alt="screen shot 2017-12-15 at 1 59 19 pm" src="https://user-images.githubusercontent.com/7240478/34060460-1c6dd94e-e1a1-11e7-9716-e21407abde82.png">

The Video and audio feature block won't appear for Personal plan sites or Jetpack plans.

Fixes: #20759

jeremeylduvall commented:
@lamosty when you're back from the holidays, would you mind taking a peek at this one?

lamosty commented on client/blocks/product-purchase-features/product-purchase-features-list/index.jsx:
```diff
@@ -85,7 +85,11 @@ class ProductPurchaseFeaturesList extends Component {
-			<VideoAudioPosts selectedSite={ selectedSite } key="videoAudioPostsFeature" />,
+			<VideoAudioPosts
+				selectedSite={ selectedSite }
+				plan="business"
```
Can we use the `PLAN_BUSINESS` constant?

lamosty commented on client/blocks/product-purchase-features/product-purchase-features-list/video-audio-posts.jsx:
```diff
@@ -13,15 +13,21 @@ import { localize } from 'i18n-calypso';
-export default localize( ( { selectedSite, translate } ) => {
+export default localize( ( { selectedSite, plan, translate } ) => {
+	const storageLimit =
+		plan === 'business'
```
Can we use the `PLAN_BUSINESS` constant instead of `business` string? You can import it with:

```
import {	PLAN_BUSINESS } from 'lib/plans/constants';
```

lamosty commented on client/blocks/product-purchase-features/product-purchase-features-list/video-audio-posts.jsx:
```diff
@@ -13,15 +13,21 @@ import { localize } from 'i18n-calypso';
+	const storageLimit =
+		plan === 'business'
+			? 'The Business plan also adds unlimited file storage.'
+			: 'The Premium plan also adds 10GB of file storage.';
```
Don't forget about i18n. Just run it through `translate`.

lamosty replied:
Actually, we can't use string concatenation with variables like this. There are a few solutions to this (see [i18n-calypso](https://github.com/Automattic/i18n-calypso#translate-method) for all options of `translate`) but it depends on whether we want to have the ability to translate the whole feature description at once or just the particular sentences.

If whole at once, just duplicate the message and change the storage text according to `plan`. If just the storage sentence, we could translate the first part of the description into one variable, then the second storage part into another variable and then concatenate them in the `description` field.

I would say putting the whole description into `translate` is preferred, even though we have to duplicate parts of the text.

jeremeylduvall commented:
@lamosty thanks for the review! I believe I addressed all of your comments. Quick question on `translate`. I pulled everything out in to a switch statement like this:

```js
let featureDescription;

switch ( plan ) {
	case PLAN_BUSINESS:
		featureDescription = translate(
			'Enrich your posts with video and audio, uploaded directly on your site. ' +
				'The Business plan also adds unlimited file storage.'
		);
		break;
	default:
		featureDescription = translate(
			'Enrich your posts with video and audio, uploaded directly on your site. ' +
				'The Premium plan also adds 10GB of file storage.'
		);
		break;
}
```

However, when I then switched my locale to Español to make sure it was translate, I still received the English translation. My concatenation is modeled after [the Strings Only example](https://github.com/Automattic/i18n-calypso#strings-only) though so I'm confused as to why it's not translating correctly.

a8ci18n commented on client/blocks/product-purchase-features/product-purchase-features-list/video-audio-posts.jsx:
```diff
@@ -12,17 +12,32 @@ import { localize } from 'i18n-calypso';
+			break;
+		default:
+			featureDescription = translate(
+				'Enrich your posts with video and audio, uploaded directly on your site. ' +
```
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Enrich your posts with video and audio, uploaded directly on your site. No ads or limits. The Premium plan also adds 10GB of file storage.' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

jeremeylduvall replied:
Note, I removed the `No ads or limits` bit because it doesn't make sense to say "No ads or limits. The Premium plan also adds 10GB of file storage" since 10GB is a limit then. I _think_ what we mean is "No ads on the video or limits on video length," which I can update. Otherwise, by itself, "No ads or limits" wasn't super clear.

lamosty replied:
Makes sense to me but I can't comment on the wording. :) Mind pinging somebody who can?

lamosty commented on client/blocks/product-purchase-features/product-purchase-features-list/video-audio-posts.jsx:
```diff
@@ -12,17 +12,32 @@ import { localize } from 'i18n-calypso';
+export default localize( ( { selectedSite, plan, translate } ) => {
+	let featureDescription;
+
+	switch ( plan ) {
```
I think it would be better if we explicitly list the possible plans instead of making a case just for Business and having Premium as default. Imagine somebody used this component for the Personal plan, totally forgetting about writing description for it. The Premium's one would be used.

`default` can be an empty text in that case. I think it's easier to notice blank description than to notice it's a description for Premium plan.

jeremeylduvall replied:
Sounds good! Updated to an empty string for `default`.

jeremeylduvall commented:
I updated the PR to have specific strings for the Business and Premium plans and set the default string to empty. Still works as intended above!

@michelleweber for the copy review, here is the original string:

> Enrich your posts with video and audio, uploaded directly on your site. No ads or limits. The Premium plan also adds 10GB of file storage.

I don't know if "No ads or limits" makes sense in that context since we talk about a limit in the next sentence. I suspect we mean "No ads on the video or limits on the length." I opted to remove that line and use:

**Business plan**

> Enrich your posts with video and audio, uploaded directly on your site. The Business plan also adds unlimited file storage.

**Premium plan**

> Enrich your posts with video and audio, uploaded directly on your site. The Premium plan also adds 10GB of file storage.

a8ci18n commented on client/blocks/product-purchase-features/product-purchase-features-list/video-audio-posts.jsx:
```diff
@@ -10,17 +10,35 @@ import { localize } from 'i18n-calypso';
+			break;
+		case PLAN_PREMIUM:
+			featureDescription = translate(
+				'Enrich your posts with video and audio, uploaded directly on your site. ' +
```
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Enrich your posts with video and audio, uploaded directly on your site. No ads or limits. The Premium plan also adds 10GB of file storage.' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/blocks/product-purchase-features/product-purchase-features-list/video-audio-posts.jsx:
```diff
@@ -10,17 +10,35 @@ import { localize } from 'i18n-calypso';
+	switch ( plan ) {
+		case PLAN_BUSINESS:
+			featureDescription = translate(
+				'Enrich your posts with video and audio, uploaded directly on your site. ' +
```
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Enrich your posts with video and audio, uploaded directly on your site. No ads or limits. The Premium plan also adds 10GB of file storage.' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

michelleweber commented:
Good call on nixing "no ads or limits;" it was definitely confusing, and I was about to say you should delete it :)

I might tweak just a little: 
Enrich your posts and pages with video or audio. Upload up to 10GB of media directly to your site.

Enrich your posts and pages with video or audio. Upload as much media as you want, directly to your site -- the Business Plan has unlimited storage.

jeremeylduvall commented:
Updated the copy per Michelle's suggestion! I think this should be ready to go if you're good with it @lamosty.

michelleweber commented:
Yup, em dash. I forget that not every site renders double hyphen as em dashes, sorry!

jeremeylduvall commented:
Sounds good! Will fix up the em dash and merge. I mixed up some commits during a rebase so I'll fix that as well :)
